package/MDAnalysis/topology/HoomdXMLParser.py

```diff
@@ -0,0 +1,156 @@
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+HOOMD XML topology parser
+========================
+Read a list of atoms from a HOOMD XML file to build a basic topology.
+
+Atom names are set to atom type if not present (which they probably aren't).
+Elements are guessed based on atom types.
+Masses and charges are set to zero if not present in the XML file.
+Hoomd XML does not identify molecules or residues, so placeholder values
+are used for residue numbers and residue names.
+Bonds and angles are read if present.
+
+Classes
+-------
+
+.. autoclass:: HoomdXMLParser
+   :members:
+   :inherited-members:
+"""
+from __future__ import absolute_import
+from ..core.AtomGroup import Atom
+from .core import guess_atom_element
+from .base import TopologyReader
+import xml.etree.ElementTree as ET
+
+class HoomdXMLParser(TopologyReader):
+    def parse(self):
+        """Parse Hoomd XML file *filename* and return the dict `structure`.
+            
+            Hoomd XML format does not contain a node for names. The parser will
+            look for a name node anyway, and if it doesn't find one, it will use
+            the atom types as names. If the Hoomd XML file doesn't contain a type
+            node (it should), then all atom types will be \'none\'. Similar to the
+            names, the parser will try to read element, mass, and charge from the XML
+            file, but it will use placeholder values if they are not present.
+
+            Because Hoomd uses unitless mass, charge, etc., if they are not present
+            they will not be guessed - they will be set to zero.
+
+            :Returns: MDAnalysis internal *structure* dict
+
+            .. SeeAlso:: The *structure* dict is defined in
+            :func:`MDAnalysis.topology.base`.
+
+        """
+        tree = ET.parse(self.filename)
```

Comment:
Is there a way to read from a stream so that you could read from compressed files?

I see that this more complicated than just using `openany` (unless you can do
```python
with lib.util.openany(self.filename) as stream:
  tree = ET.parse(stream)
root = tree.getroot()
configuration = root.find('configuration')
...
```
in which case it would be easy ;-) ).

If it's not easy then just ignore my previous comment on compressing everything and leave it uncompressed.